synapse/http/server.py

```diff
@@ -809,3 +830,24 @@ def finish_request(request: Request):
         request.finish()
     except RuntimeError as e:
         logger.info("Connection disconnected before response was written: %r", e)
```

Comment:
From https://github.com/matrix-org/synapse/pull/10844#discussion_r713032338

> is this really the only possible reason that a RuntimeError can be raised?

Probably not, but this is what it was before.